odoo/tools/yaml_tag.py

```diff
@@ -149,16 +149,24 @@ def ref_constructor(loader, tag_suffix, node):
 # Constructors are actually defined globally: do not redefined them in another
 # class/file/package.  This means that module recorder need import this file.
 def add_constructors():
-    yaml.add_constructor(u"!assert", assert_constructor)
-    yaml.add_constructor(u"!record", record_constructor)
-    yaml.add_constructor(u"!python", python_constructor)
-    yaml.add_constructor(u"!menuitem", menuitem_constructor)
-    yaml.add_constructor(u"!act_window", act_window_constructor)
-    yaml.add_constructor(u"!function", function_constructor)
-    yaml.add_constructor(u"!report", report_constructor)
-    yaml.add_constructor(u"!context", context_constructor)
-    yaml.add_constructor(u"!delete", delete_constructor)
-    yaml.add_constructor(u"!url", url_constructor)
-    yaml.add_constructor(u"!eval", eval_constructor)
-    yaml.add_multi_constructor(u"!ref", ref_constructor)
+    constructors = [
+        ("!assert", assert_constructor),
+        ("!record", record_constructor),
+        ("!python", python_constructor),
+        ("!menuitem", menuitem_constructor),
+        ("!act_window", act_window_constructor),
+        ("!function", function_constructor),
+        ("!report", report_constructor),
+        ("!context", context_constructor),
+        ("!delete", delete_constructor),
+        ("!url", url_constructor),
+        ("!eval", eval_constructor),
+        ("!ref", ref_constructor),
+    ]
+    for tag, constructor in constructors:
+        try:
+            # pyYAML v5.1 requires a loader to be set and does not include a default for add_constructor.
+            yaml.add_constructor(tag, constructor, Loader=yaml.FullLoader)
```

Comment:
Restricting the code in the try block to just determining if yaml.FullLoader is available eliminates any AttributeError escaping from the add_constructor call as a potential reason for executing the except block. The add_constructor with Loader can then be in an else.